### PR TITLE
Unit tests for `node` package + small refactoring & fixes

### PR DESCRIPTION
**Pull request by iulianpascalau**
## Reasoning behind the pull request
- added unit tests for increasing code coverage
  
## Proposed changes
- added unit tests for the node runner

## Testing procedure
- standard system test. Stopping the node & shuffling out should work, also, the `stats` directory should now contain the `genesisSmartContracts.json` and the `gasSchedules` directory and all the gas schedules files.

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: node/mock/applicationRunningTrigger.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+// NewApplicationRunningTrigger -
+func NewApplicationRunningTrigger() *applicationRunningTrigger {
+	return &applicationRunningTrigger{
+		chanClose: make(chan struct{}),
+	}
+}
+
+func (trigger *applicationRunningTrigger) Write(p []byte) (n int, err error) {
```
Comment (iulianpascalau) [inline node/mock/applicationRunningTrigger.go:22]: missing comment

File: node/nodeRunner_test.go [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+	file4Contents := []byte("file4")
+
+	tempDir := t.TempDir()
+
+	// generating dummy structure like
+	// file1
+	// src
+	//   +- file2
+	//   +- dir1
+	//	       +- file3
```
Comment (iulianpascalau) [inline node/nodeRunner_test.go:184]: indentation error

File: node/mock/applicationRunningTrigger.go (hunk 1 above)
Comment (iulianpascalau) [reply node/mock/applicationRunningTrigger.go:22]: added, thanks

File: node/nodeRunner_test.go (hunk 2 above)
Comment (iulianpascalau) [reply node/nodeRunner_test.go:184]: fixed

**Review by raduchis — APPROVED** [review]

File: node/metrics/metrics_test.go [hunk 3]
```diff
@@ -329,3 +332,185 @@ func TestInitRatingsMetrics(t *testing.T) {
...
+			common.MetricMinTransactionVersion:        uint64(1),
+			common.MetricNumValidators:                uint64(2),
+			common.MetricConsensusGroupSize:           uint64(63),
+		}
+
+		assert.Equal(t, len(expectedValues), len(keys))
+		for k, v := range expectedValues {
+			assert.Equal(t, v, keys[k], fmt.Sprintf("for key %s", k))
+		}
+	})
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply node/metrics/metrics_test.go:471]: added, thanks

File: node/metrics/metrics_test.go [hunk 4]
```diff
@@ -329,3 +332,185 @@ func TestInitRatingsMetrics(t *testing.T) {
...
+			return 111111
+		},
+	}
+	version := "version"
+	economicsConfigs := &config.EconomicsConfig{
+		RewardsSettings: config.RewardsSettings{
+			RewardsConfigByEpoch: []config.EpochRewardSettings{
+				{
+					LeaderPercentage: 2,
+				},
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply node/metrics/metrics_test.go:387]: added

File: node/metrics/metrics_test.go [hunk 5]
```diff
@@ -329,3 +332,185 @@ func TestInitRatingsMetrics(t *testing.T) {
 		assert.Equal(t, v, keys[k])
 	}
 }
+
+func TestInitMetrics(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply node/metrics/metrics_test.go:336]: added, thanks.

File: node/nodeRunner_test.go [hunk 6]
```diff
@@ -0,0 +1,437 @@
...
+			stopChan,
+			healthServiceClosableComponent,
+			facadeClosableComponent,
+			httpClosableComponent,
+			n,
+			1,
+		)
+
+		// these exceptions appear because the delayedComponent prevented the call of the first 2 components
+		// as the closable components are called in revered order
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply node/nodeRunner_test.go:377]: changed

**Review by raduchis — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed. 
@@                            Log scanner                                  @@
##                            add-unit-tests-for-node-package              ##
================================================================================
- Known Warnings              14
- New Warnings                2
- Known Errors                0
- New Errors                  1
- Panics                      0
================================================================================
- block hash does not match                  10938
- wrong nonce in block                       3558
- miniblocks does not match                  0
- num miniblocks does not match              0
- miniblock hash does not match              0
- block bodies does not match                0
- receipts hash missmatch                    0
================================================================================
-  No jailed nodes on the thestnet
================================================================================

**Review by iulianpascalau — COMMENTED** [review]
Refactored some tests due to this comment: https://github.com/multiversx/mx-chain-go/pull/5141#discussion_r1154190191

File: node/external/timemachine/fee/feeComputer_test.go [hunk 7]
```diff
@@ -216,3 +216,18 @@ func TestFeeComputer_InHighConcurrency(t *testing.T) {
 
 	wg.Wait()
 }
+
+func TestFullHistoryPruningStorer_IsInterfaceNil(t *testing.T) {
```
Comment (sstanculeanu) [inline node/external/timemachine/fee/feeComputer_test.go:220]:
```suggestion
func TestFeeComputer_IsInterfaceNil(t *testing.T) {
```

Comment (iulianpascalau) [reply node/external/timemachine/fee/feeComputer_test.go]: too much copy-paste :D
Thanks.

**Review by sstanculeanu — DISMISSED** [review]

File: node/external/logs/logsFacade_test.go [hunk 8]
```diff
@@ -144,3 +144,18 @@ func TestLogsFacade_IncludeLogsInTransactionsShouldWork(t *testing.T) {
 	require.Nil(t, transactions[2].Logs)
 	require.Equal(t, "fourth", transactions[3].Logs.Events[0].Identifier)
 }
+
+func TestLogsFacade_IsInterfaceNil(t *testing.T) {
+	t.Parallel()
+
+	var lf *logsFacade
+	require.True(t, check.IfNil(lf))
```
Comment (raduchis) [inline node/external/logs/logsFacade_test.go:152]:
```suggestion
	require.True(t, lf.IsInterfaceNil())
```

File: node/external/logs/logsFacade_test.go [hunk 9]
```diff
@@ -144,3 +144,18 @@ func TestLogsFacade_IncludeLogsInTransactionsShouldWork(t *testing.T) {
...
+
+	arguments := ArgsNewLogsFacade{
+		StorageService:  genericMocks.NewChainStorerMock(7),
+		Marshaller:      &marshal.GogoProtoMarshalizer{},
+		PubKeyConverter: testscommon.NewPubkeyConverterMock(32),
+	}
+	lf, _ = NewLogsFacade(arguments)
+	require.False(t, check.IfNil(lf))
```
Comment (raduchis) [inline node/external/logs/logsFacade_test.go:160]:
```suggestion
	require.False(t, lf.IsInterfaceNil())
```

File: node/external/logs/logsFacade_test.go (hunk 8 above)
Comment (iulianpascalau) [reply node/external/logs/logsFacade_test.go]: done on all package

**Review by sstanculeanu — APPROVED** [review]

File: node/external/logs/logsFacade_test.go (hunk 8 above)
Comment (raduchis) [reply node/external/logs/logsFacade_test.go]: ok!

**Review by raduchis — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]

**Pull request merged**